bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/network/MagentaTVNetwork.java

```diff
@@ -0,0 +1,178 @@
+                StringBuilder sb = new StringBuilder();
+                for (int i = 0; i < mac.length; i++) {
+                    sb.append(String.format("%02X%s", mac[i], (i < mac.length - 1) ? ":" : ""));
+                }
+                localMAC = sb.toString().toUpperCase();
+                logger.debug("Local IP address={}, Local MAC address = {}", localIP, localMAC);
+                return;
+            }
+        } catch (UnknownHostException | SocketException e) {
+            throw new MagentaTVException(e);
+        }
+
+        throw new MagentaTVException(
+                "Unable to get local IP / MAC address, check network settings in openHAB system configuration!");
+    }
+
+    /**
+     * Checks if client ip equals or is in range of ip networks provided by
+     * semicolon separated list
+     *
+     * @param clientIp in numeric form like "192.168.0.10"
+     * @param ipList like "127.0.0.1;192.168.0.0/24;10.0.0.0/8"
+     * @return true if client ip from the list os ips and networks
+     */
+    public static boolean isIpInSubnet(String clientIp, String ipList) {
+        if (ipList.isEmpty()) {
+            // No ip address provided
+            return true;
+        }
+        String[] subnetMasks = ipList.split(";");
+        for (String subnetMask : subnetMasks) {
+            subnetMask = subnetMask.trim();
+            if (clientIp.equals(subnetMask)) {
+                return true;
+            }
+            if (subnetMask.contains("/")) {
+                if (new SubnetUtils(subnetMask).getInfo().isInRange(clientIp)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    @Nullable
+    public NetworkInterface getLocalInterface() {
+        return localInterface;
+    }
+
+    public String getLocalIP() {
+        return localIP;
+    }
+
+    public String getLocalPort() {
+        return localPort;
+    }
+
+    public String getLocalMAC() {
+        return localMAC;
+    }
+
+    public static final int WOL_PORT = 9;
+
+    /**
+     * Send a Wake-on-LAN packet
+     *
+     * @param ipAddr destination ip
+     * @param macAddress destination MAC address
+     * @throws MagentaTVException
+     */
+    public void sendWakeOnLAN(String ipAddr, String macAddress) throws MagentaTVException {
+        try {
+            byte[] macBytes = getMacBytes(macAddress);
+            byte[] bytes = new byte[6 + 16 * macBytes.length];
+            for (int i = 0; i < 6; i++) {
+                bytes[i] = (byte) 0xff;
+            }
+            for (int i = 6; i < bytes.length; i += macBytes.length) {
+                System.arraycopy(macBytes, 0, bytes, i, macBytes.length);
+            }
+
+            InetAddress address = InetAddress.getByName(ipAddr);
+            DatagramPacket packet = new DatagramPacket(bytes, bytes.length, address, WOL_PORT);
+            DatagramSocket socket = new DatagramSocket();
```

Comment:
Maybe use try-with-resources to auto close:
```suggestion
            try (DatagramSocket socket = new DatagramSocket()) {
```

Reply:
changed

this needs no socket.close(), correct?

Reply:
correct. When using, what is called, try-with-resources java will automatically close the stream. This works for all code that implements `Closable`.